#============================================================================================================
# Resolve issue with Clang 13, due to warning option 'unused-but-set-variable' not being available.
# Fixed by upstream, and will be included in next release.
#
# Details:
# - https://github.com/pytorch/pytorch/issues/74805
# - https://github.com/pytorch/pytorch/pull/80916
#============================================================================================================
--- a/cmake/Dependencies.cmake	2022-07-17 17:29:02.000000000 -0400
+++ b/cmake/Dependencies.cmake	2022-07-17 17:30:28.000000000 -0400
@@ -817,8 +817,13 @@
     set_property(TARGET fbgemm_avx512 PROPERTY POSITION_INDEPENDENT_CODE ON)
     set_property(TARGET fbgemm PROPERTY POSITION_INDEPENDENT_CODE ON)
     if("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 13.0.0)
-        # See https://github.com/pytorch/pytorch/issues/74352
-        target_compile_options(asmjit PRIVATE -Wno-deprecated-copy -Wno-unused-but-set-variable)
+      # See https://github.com/pytorch/pytorch/issues/74352
+      target_compile_options(asmjit PRIVATE -Wno-deprecated-copy)
+      if(("${CMAKE_CXX_COMPILER_ID}" STREQUAL "AppleClang" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 13.1.6)
+        OR("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 13.0.0))
+        # -Wno-unused-but-set-variable doesn't exist in Apple clang version 13.0.0 (clang-1300.0.29.30)
+        target_compile_options(asmjit PRIVATE -Wno-unused-but-set-variable)
+      endif()
     endif()
   endif()
 
